pkg/kv/iterators.go

```diff
@@ -253,24 +253,29 @@ type PartitionIterator struct {
 }
 
 func NewPartitionIterator(ctx context.Context, store Store, msgType protoreflect.MessageType, partitionKey string) (*PartitionIterator, error) {
-	itr, err := store.Scan(ctx, []byte(partitionKey), []byte(""))
-	if err != nil {
-		return nil, err
-	}
 	return &PartitionIterator{
 		ctx:          ctx,
 		store:        store,
 		msgType:      msgType,
-		itr:          itr,
 		partitionKey: partitionKey,
 	}, nil
 }
 
+func getNewIter(ctx context.Context, store Store, partitionKey string, start []byte) (EntriesIterator, error) {
```

Comment:
Not sure we need a function that calls a function - if it was a member to capture the use of: p.{ctx, store, paritionKey} it was a different story.